src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -1094,6 +1094,24 @@ int dhcp_server_handle_message(sd_dhcp_server *server, DHCPMessage *message, siz
         existing_lease = hashmap_get(server->bound_leases_by_client_id, &req->client_id);
         static_lease = hashmap_get(server->static_leases_by_client_id, &req->client_id);
 
+        /* when no lease is found based on the client id fall back to chaddr */
+        if (!static_lease) {
```

Comment:
The function is already long. Please split-out the logic as a function, e.g. something like the following:
```c
static DHCPLease* server_get_static_lease(sd_dhcp_server *server, const DHCPRequest *req) {
        ...

        static_lease = hashmap_get(...);
        if (static_lease)
                return static_lease;

        c = (DHCPClientId) {
                ...,
        };

        return hashmap_get(...);
}
```